compiler/symbol.go

```diff
@@ -218,6 +218,9 @@ func (c *compilerContext) getFunctionInfo(f *ssa.Function) functionInfo {
 		// Pick the default linkName.
 		linkName: f.RelString(nil),
 	}
+	if isWrapperFunction(f) {
+		info.linkName += "$" + f.Synthetic
+	}
```

Comment:
Can you extract this and move it into a separate PR with tests?
You can test this by putting the Go code that led to the bug in one of the compiler/testdata/*.go files and running `go test ./compiler -update`.

Reply:
Thanks for the feedback! I will isolate the compiler fix in a separate PR and attempt to create a test case that reproduces the issue 👍